init_parameters.txt
% utility parameters (alpha)
0.59965   % CRRA income parameter (1)
197 % 9.76866   % women utility from children when married (2)
8.64472   % women utility from children when single (3)
295 % 5.94454   % men utility from children when maarried (4)
31.86416  % utility from leisure as a function of number of children (5)
257.76214 % utility from leisure (6)
% wage parameters wife (beta)
0.0415013   % experience (7)
0.051606   % experience (8)
0.0586301   % experience (9)
0.0604871   % experience (10)
-0.0005   % exp^2 (11)
-0.0008   % exp^2 (12)
-0.0007   % exp^2 (13)
-0.0013   % exp^2 (14)
9.681104   % HSG - schooling (15)
9.89622   % SC -  schooling (16)
10.16739  % CG -  schooling (17)
10.44239848  % PC -  schooling (18)
% wage parameters husband (beta)
0.04917   % experience (19)
0.06671   % experience (20)
0.07869   % experience (21)
0.08693   % experience (22)
0.099893   % experience (23)
-0.00123  % exp^2 (24)
-0.00143  % exp^2 (25)
-0.00163  % exp^2 (26)
-0.00177  % exp^2 (27)
-0.00101  % exp^2 (28)
9.91566   % HSD - schooling (29)
9.91285   % HSG - schooling (30)
9.98839   % SC - schooling (31)
10.19046  % CG - schooling (32)
10.4973603  % PC - schooling (33)
% job offer parameters wife (row)
2.69671   % work in previous period (34)
0.33691   % HSG - schooling (35)
0.36611   % SC - schooling (36)
0.39732   % CG - schooling (37)
0.41509311   % PC - schooling (38)
0.01501   % work experience (39)
% home production technology - parameters (hp)
1.2342786   % economies of scale in marriage (40)
% probability of another child parameters (c)
0000000000   % canceled parameter (41)
-0.0664251518  % age wife - HSG (42)
-0.0000003  % age square wife - HSG (43)
-0.06461917  % age wife - SC (44)
0.00000   % age square wife - SC (45)
-0.06256283  % age wife - CG (46)
-.00011  % age square wife - CG (47)
-0.0596356369  % age wife - PC (48)
-0.000003  % age square wife - PC (49)
-0.088953952  % number of children at household (50)
0.001099   % schooling - husband (51)
0.918396   % married (52)
% terminal value Parameters (t)
10.66081  % schooling wife - HSG (53)
18.94146  % schooling wife - SC (54)
31.20066  % schooling wife - CG (55)
44.32608  % schooling wife - PC (56)
13.46640  % exp wife (57)
7.76321   % schooling husband if married - HSD (58)
12.93352  % schooling husband if married - HSG (59)
17.08230  % schooling husband if married - SC (60)
139.33771  % schooling husband if married - CG (61)
131.06585  % schooling husband if married - PC (62)
6.93452   % exp husband if married (63)
3923.24595  % martial status (64)
3.82900   % number of children (65)
16.04158  % match quality if married (66)
2.03147   % number of children if married (67)
21.40492  % previous work state - wife (68)
0.0       % dummy (69)
6.76035   % schooling wife if married- HSG (70)
8.44978   % schooling wife if married- SC (71)
13.54135  % schooling wife if married- CG (72)
586.57359  % schooling wife if married- PC (73)
8.52153   % exp wife if married (74)
-8.48603  % schooling husband - HSD (75)
2.19182   % schooling husband - HSG (76)
21.45374  % schooling husband - SC (77)
37.67797  % schooling husband - CG (78)
48.15956  % schooling husband - PC (79)
8.69768   % exp husband (80)
2934.00441  % martial status if married (81)
1.92952   % number of children if married (82)
11.59128  % match quality if married (83)
3.02324   % number of children if married - NO NEED (84)
3.57758   % previous work state - wife if married (85)
0.0       % dummy (86)
% probability of meeting a partner (p)
-1.672575499482  % probability of meeting a potential husband - wife constant (87)
0.08512   % probability of meeting a potential husband - wife age (88)
-0.00280  % probability of meeting a potential husband - wife age square (89)
-0.83845  % probability of meeting a potential husband - husband constant (90)
0.09415   % probability of meeting a potential husband - husband age (91)
-0.00237  % probability of meeting a potential husband - husband age square (92)
-1.67008  % probability of match quality decrease (93)
-0.68139713  % probability of match quality increase (94)
15.47873  % education match parameter (95)
15.33525  % education match parameter (96)
15.88097  % education match parameter (97)
1.54916  % education match parameter (98)
% random shocks variance-covariance matrix (sigma)
-0.97272  % wage error variance husband (99)
-1.11858  % wage error variance wife (100)
-1.08414  % variance wife ability (101)
-0.94988  % variance husband ability (102)
-1.16801  % variance match quality (103)


init_parameters_new.txt
301.5445
57.3595
57.3595
17.3656
17.3656
66.3665
66.3665
0.0529
-0.0012
9.4822
9.6358
10.0558
10.4458
0.0635
-0.0013
9.4551
9.6371
9.8054
10.1379
10.3943
3.8958
-0.8054
-0.6549
-0.4480
-0.0926
-0.0283
51.8700
38.3885
850.5489
7.2600
7.3269
0.7824
0.9412
-0.0023
-0.0875
0.0004
-0.0885
0.0007
-0.0883
0.0008
-0.0589
-0.0003
-0.0396
0.1157
0.9037
12.2542
17.8522
22.2458
43.8854
15.5145
-3.2554
11.8144
14.4414
15.7427
13.8574
7.5475
125.8554
6.8551
20.9658
2.1148
20.8657
2.7885
4.3552
4.9556
6.5841
3.5588
-20.5841
0.00
33.5844
54.7589
58.6952
17.8544
92.5884
3.8557
14.5880
0.00
-0.4566
-1.2489
-1.1257
612.2500
108.7552
791.3645
783.2600
-0.5151
-0.8147
-0.7456
-0.7236
0.8796





c++/parameters.h
#pragma once

#include <array>
#include <vector>
#include <cmath>
#include "matrix.h"

// TODO auto-generate these numbers based on the actual files
const size_t PARAMETER_LIST_LEN = 103;
const size_t TAX_ROW = 56;
const size_t TAX_COL = 43;
const size_t DED_COL = 30;
const size_t WIVES_ROW = 47;
const size_t WIVES_COL = 41;
const size_t HUSBANDS_ROW = 33;
const size_t HUSBANDS_COL = 26;


using ParameterList = std::array<double, PARAMETER_LIST_LEN>;
using TaxBrackets = Matrix<TAX_ROW, TAX_COL>;
using DeductionExceptions = Matrix<TAX_ROW, DED_COL>;
using Wives = Matrix<WIVES_ROW, WIVES_COL>;
using Husbands = Matrix<HUSBANDS_ROW, HUSBANDS_COL>;

struct Parameters {
  // utility parameters
  const double alpha;          // CRRA income parameter 
  const double alpha1_w_m;     // women utility from children when married
  const double alpha1_w_s;     // women utility from children when single
  const double alpha1_h_m;     // men utility from children when maarried
  const double alpha2_w;       // utility from leisure as a function of number of children
  const double alpha3_w;       // utility from leisure
  // wage parameters wife
  const double beta11_w;       // experience 
  const double beta12_w;       // experience 
  const double beta13_w;       // experience 
  const double beta14_w;       // experience 
  const double beta21_w;       // exp^2  
  const double beta22_w;       // exp^2  
  const double beta23_w;       // exp^2  
  const double beta24_w;       // exp^2  
  const double beta31_w;       // HSG - schooling    
  const double beta32_w;       // SC -  schooling    
  const double beta33_w;       // CG -  schooling    
  const double beta34_w;       // PC -  schooling
  // wage parameters husband
  const double beta10_h;       // experience 
  const double beta11_h;       // experience 
  const double beta12_h;       // experience 
  const double beta13_h;       // experience 
  const double beta14_h;       // experience 
  const double beta20_h;       // exp^2
  const double beta21_h;       // exp^2
  const double beta22_h;       // exp^2
  const double beta23_h;       // exp^2
  const double beta24_h;       // exp^2
  const double beta30_h;       // HSD - schooling    
  const double beta31_h;       // HSG- schooling 
  const double beta32_h;       // SC -  schooling    
  const double beta33_h;       // CG -  schooling    
  const double beta34_h;       // PC -  schooling    
  // job offer parameters - wife 
  const double row0_w;         // work in previous period
  const double row11_w;        // HSG -schooling
  const double row12_w;        // SC - schooling
  const double row13_w;        // CG - schooling
  const double row14_w;        // PC - schooling
  const double row2_w;         // work experience
  // home production technology - parameters    
  const double hp;             // sigma CRRA parameter
  // probability of another child parameters    
  const double c1;             // previous work state - wife
  const double c2;             // age wife - HSG
  const double c3;             // age square wife - HSG
  const double c4;             // age wife - SC
  const double c5;             // age square wife - SC
  const double c6;             // age wife - CG
  const double c7;             // age square wife - CG
  const double c8;             // age wife - PC
  const double c9;             // age square wife - PC
  const double c10;            // number of children at household    
  const double c11;            // schooling - husband
  const double c12;            // married - M
  // terminal value parameters
  const double t1_w;           // schooling wife - HSG
  const double t2_w;           // schooling wife - SC
  const double t3_w;           // schooling wife - CG
  const double t4_w;           // schooling wife - PC
  const double t5_w;           // exp wife
  const double t6_w;           // schooling husband if married - HSD
  const double t7_w;           // schooling husband if married - HSG
  const double t8_w;           // schooling husband if married - SC
  const double t9_w;           // schooling husband if married - CG
  const double t10_w;          // schooling husband if married - PC
  const double t11_w;          // exp husband if married 
  const double t12_w;          // martial status
  const double t13_w;          // number of children
  const double t14_w;          // match quality if married
  const double t15_w;          // number of children if married
  const double t16_w;          // previous work state - wife
  const double t17_w;          // previous work state - husband if married
  // terminal value Parameters
  const double t1_h;           // schooling wife if married- HSG
  const double t2_h;           // schooling wife if married- SC
  const double t3_h;           // schooling wife if married- CG
  const double t4_h;           // schooling wife if married- PC
  const double t5_h;           // exp wife if married
  const double t6_h;           // schooling husband - HSD
  const double t7_h;           // schooling husband - HSG
  const double t8_h;           // schooling husband - SC
  const double t9_h;           // schooling husband - CG
  const double t10_h;          // schooling husband - PC
  const double t11_h;          // exp husband  
  const double t12_h;          // martial status if married
  const double t13_h;          // number of children if married
  const double t14_h;          // match quality if married
  const double t15_h;          // number of children if married - NO NEED
  const double t16_h;          // previous work state - wife if married
  const double t17_h;          // previous work state - husband 
  // match quality parameters
  const double p0_w;           // probability of meeting a potential husband - wife constant
  const double p1_w;           // probability of meeting a potential husband - wife age
  const double p2_w;           // probability of meeting a potential husband - wife age square
  const double p0_h;           // probability of meeting a potential husband - husband constant
  const double p1_h;           // probability of meeting a potential husband - husband age
  const double p2_h;           // probability of meeting a potential husband - husband age square
  const double MATCH_Q_DECREASE;   // probability of match quality decrease
  const double MATCH_Q_INCREASE;   // probability of match quality increase
  const std::array<double,5> EDUC_MATCH;  // education match parameters
  const std::array<double,5> sigma;       // diagonal of random shocks variance-covariance matrix
  const TaxBrackets tax;
  const DeductionExceptions ded;
  const Wives wives;
  const Husbands husbands1;
  const Husbands husbands2;
  const Husbands husbands3;
  const Husbands husbands4;
  const Husbands husbands5;

  // initialization of parameters from list
  Parameters(const ParameterList& parameter_list, 
      const TaxBrackets& _tax, 
      const DeductionExceptions& _ded, 
      const Wives& _wives,
      const Husbands _husbands1,
      const Husbands _husbands2,
      const Husbands _husbands3,
      const Husbands _husbands4,
      const Husbands _husbands5) : 
    alpha(parameter_list[0]),
    alpha1_w_m(parameter_list[1]),
    alpha1_w_s(parameter_list[2]),
    alpha1_h_m(parameter_list[3]),
    alpha2_w(parameter_list[4]),
    alpha3_w(parameter_list[5]),
    beta11_w(parameter_list[6]),
    beta12_w(parameter_list[7]),
    beta13_w(parameter_list[8]),
    beta14_w(parameter_list[9]),
    beta21_w(parameter_list[10]),
    beta22_w(parameter_list[11]),
    beta23_w(parameter_list[12]),
    beta24_w(parameter_list[13]),
    beta31_w(parameter_list[14]),
    beta32_w(parameter_list[15]),
    beta33_w(parameter_list[16]),
    beta34_w(parameter_list[17]),
    beta10_h(parameter_list[18]),
    beta11_h(parameter_list[19]),
    beta12_h(parameter_list[20]),
    beta13_h(parameter_list[21]),
    beta14_h(parameter_list[22]),
    beta20_h(parameter_list[23]),
    beta21_h(parameter_list[24]),
    beta22_h(parameter_list[25]),
    beta23_h(parameter_list[26]),
    beta24_h(parameter_list[27]),
    beta30_h(parameter_list[28]),
    beta31_h(parameter_list[29]),
    beta32_h(parameter_list[30]),
    beta33_h(parameter_list[31]),
    beta34_h(parameter_list[32]),
    row0_w(parameter_list[33]),
    row11_w(parameter_list[34]),
    row12_w(parameter_list[35]),
    row13_w(parameter_list[36]),
    row14_w(parameter_list[37]),
    row2_w(parameter_list[38]),
    hp(parameter_list[39]),
    c1(parameter_list[40]),
    c2(parameter_list[41]),
    c3(parameter_list[42]),
    c4(parameter_list[43]),
    c5(parameter_list[44]), 
    c6(parameter_list[45]),
    c7(parameter_list[46]),
    c8(parameter_list[47]),
    c9(parameter_list[48]),
    c10(parameter_list[49]),
    c11(parameter_list[50]),
    c12(parameter_list[51]),
    t1_w(parameter_list[52]),
    t2_w(parameter_list[53]),
    t3_w(parameter_list[54]),
    t4_w(parameter_list[55]),
    t5_w(parameter_list[56]),
    t6_w(parameter_list[57]),
    t7_w(parameter_list[58]),
    t8_w(parameter_list[59]),
    t9_w(parameter_list[60]),
    t10_w(parameter_list[61]),
    t11_w(parameter_list[62]),
    t12_w(parameter_list[63]),
    t13_w(parameter_list[64]),
    t14_w(parameter_list[65]),
    t15_w(parameter_list[66]),
    t16_w(parameter_list[67]),
    t17_w(parameter_list[68]),
    t1_h(parameter_list[69]),
    t2_h(parameter_list[70]),
    t3_h(parameter_list[71]),
    t4_h(parameter_list[72]),
    t5_h(parameter_list[73]),
    t6_h(parameter_list[74]),
    t7_h(parameter_list[75]),
    t8_h(parameter_list[76]),
    t9_h(parameter_list[77]),
    t10_h(parameter_list[78]),
    t11_h(parameter_list[77]),
    t12_h(parameter_list[80]),
    t13_h(parameter_list[81]),
    t14_h(parameter_list[82]),
    t15_h(parameter_list[83]),
    t16_h(parameter_list[84]),
    t17_h(parameter_list[85]),
    p0_w(parameter_list[86]),
    p1_w(parameter_list[87]),
    p2_w(parameter_list[88]),
    p0_h(parameter_list[89]),
    p1_h(parameter_list[90]),
    p2_h(parameter_list[91]),
    MATCH_Q_DECREASE(exp(parameter_list[92])/(1.0+exp(parameter_list[92]))),
    MATCH_Q_INCREASE(exp(parameter_list[93])/(1.0+exp(parameter_list[93]))),
    EDUC_MATCH{0.0, parameter_list[94], parameter_list[95], parameter_list[96], parameter_list[97]},
    sigma{exp(parameter_list[98]), exp(parameter_list[99]), exp(parameter_list[100]), exp(parameter_list[101]), exp(parameter_list[102])},
    tax(_tax),
    ded(_ded),
    wives(_wives),
    husbands1(_husbands1),
    husbands2(_husbands2),
    husbands3(_husbands3),
    husbands4(_husbands4),
    husbands5(_husbands5)
    {}
};

